Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.m

```diff
@@ -47,12 +47,21 @@ + (NSString *)vok_entityName
             
             // Since we don't have an entityName class method, look up the entity name in the managed object model.
             NSManagedObjectModel *model = [[VOKCoreDataManager sharedInstance] managedObjectModel];
-            for (NSEntityDescription *description in model.entities) {
-                if ([self isSubclassOfClass:NSClassFromString(description.managedObjectClassName)]) {
-                    vok_entityName = description.name;
-                    break;
+            
+            // Start with the class we are...
+            Class workingClass = self;
+            do {
+                // ... check for a matching entity in the model...
+                for (NSEntityDescription *description in model.entities) {
+                    if ([workingClass class] == [NSClassFromString(description.managedObjectClassName) class]) {
```

Comment:
So like `NSString *workingClassName = NSStringFromClass(workingClass);` at the top of the `do { … }` block and then compare `workingClassName` to `description.managedObjectClassName` using `isEqualToString:`?